fun/connect4System/reactionListener.go.tmpl

```diff
@@ -0,0 +1,194 @@
+{{/* 
+    This is the reaction listener for the connect4 system
+
+    Trigger Type: Reaction
+    Trigger: Added Reactions Only
+    
+    Credits: 
+    zen | ゼン (https://github.com/z3nn13)
+*/}}
+
+{{ define "board_maker" }}
+        {{- $store := .store -}}
+        {{- $board := cslice.AppendSlice .board -}}
+        {{- $out := "" -}}
+        {{- range $board -}}
+                {{$rows := ""}}
+                {{- range . -}}
+                        {{- $rows = printf "%s%s " $rows (str . | $store.emojis.Get) -}}
+                {{- end -}}
+                {{- $out = print $rows "\n" $out -}}
+        {{- end -}}
+        {{- $out = print $out "1️⃣ 2️⃣ 3️⃣ 4️⃣ 5️⃣ 6️⃣ 7️⃣" -}}
+        {{- $embed := sdict "author" (sdict "name" (print .cPlayer.Username "'s turn") "icon_url" (.cPlayer.AvatarURL "256")) 
+            "title" "Connect4" 
+            "description" $out
+            "color" (str .turn | $store.color.Get)
+            "fields" (cslice 1 2)
+            "footer" (sdict "text" "Powered by • Yagpdb.xyz" ) -}}
+        {{- $embed.fields.Set (sub .turn 1) (sdict "name" (print "Player " .turn) "value" (print "> " .cPlayer.Mention) "inline" true) -}}
+        {{- $embed.fields.Set (sub .nextTurn 1 ) (sdict "name" (print "Player " .nextTurn) "value" .nPlayer.Mention "inline" true ) -}}
+        {{- .Set "embed" $embed -}}
+{{- end}}
+
+{{/* Global variables */}}
+{{ $data := sdict}}{{$players := cslice}}{{$cPlayer := ""}}{{$turn := ""}}{{$nextTurn := ""}}{{$nPlayer := ""}}{{$input := ""}}{{$flag := false}}
+{{ $store := sdict
+    "emojis" (sdict "0" "⚫" "1" "🔴" "2" "🟡" "10" "🔵") 
```

Comment:
out of curiosity, why are the offsets 0, 1, 2, and 10 rather than sequential 0 1 2 3?
if you change it to be sequential you can use a cslice instead as its indices go from 0 to N.

also: assuming that 0, 1, 2, 10 represent "unused", "first player", "second player", "win" respectively, you could use an enum-like value to make the code easier to comprehend:
```
{{$SquareKind := sdict "Unused" 0 "FirstPlayer" 1 "SecondPlayer" 2 "WinningSquare" 2}}
{{$emoji := index $store.emojis $SquareKind.Unused}}
```

Reply:
> out of curiosity, why are the offsets 0, 1, 2, and 10 rather than sequential 0 1 2 3?   

That's a good question xd, in my mind I thought 3 would be harder to distinguish than a 10, but in hindsight, it would've better to be a 3 so I could use slices to my advantage as you said.

> also: assuming that 0, 1, 2, 10 represent "unused", "first player", "second player", "win" respectively, you could use an enum-like value to make the code easier to comprehend:

"0,1,2,10" represents "empty/⚫", "first player's token/🔴", "second player's token/🟡", "empty/🔵" respectively,
"10" is used for highlighting a drop with blue

![image](https://user-images.githubusercontent.com/74445896/122324150-e1f70280-cf5a-11eb-8ab1-a6aa33db959a.png)

The main problem with using an enum-like value is that $turn is not **constant**:
```go
{{ $turn := $data.turn }} {{/* starts as 1 (or) 1st player's turn */}}
{{ $nextTurn := sub 3 $turn }} {{/* nextTurn is 2 */}}
{{ $data.Set "turn" $nextTurn }}
{{ dbSet 2021 "connect4" $data }}
{{/* this alternates $turn between 1 and 2 */}}
```
Also in case you might be wondering why I'm not using a "0" and "1" instead of "1 and 2", the turn variable is used to set the token as well. If it was 0, it'd be conflicting with the "empty/black" placeholder.

Reply:
fair enough, i feel the design here could be a bit better but not gonna block this PR just for that : ) feel free to ignore this particular suggestion then